syncer/syncer.go

```diff
@@ -1404,16 +1436,37 @@ func (s *Syncer) syncDML(
 
 // Run starts running for sync, we should guarantee it can rerun when paused.
 func (s *Syncer) Run(ctx context.Context) (err error) {
-	tctx := s.tctx.WithContext(ctx)
+	runCtx, runCancel := context.WithCancel(context.Background())
+	tctx := s.tctx.WithContext(runCtx)
 
 	defer func() {
 		if s.done != nil {
 			close(s.done)
 		}
 	}()
 
+	go func() {
+		<-ctx.Done()
```

Comment:
if `runCtx` is also done here, e.g. we met an unretryable error, we needn't do the following works any more

Reply:
fixed in https://github.com/pingcap/dm/pull/1928/commits/b2d6355f9dfe3f5241bde6edbe5f4aaad280570b